src/main/kotlin/com/hypto/iam/server/security/Authorization.kt

```diff
@@ -171,15 +172,7 @@ fun getResourceHrnFunc(
     }
 }
 
-data class HrnTemplateInput(
-    val pathTemplate: String,
-    val resourceNameIndex: Int,
-    val resourceInstanceIndex: Int,
-    val organizationIdIndex: Int,
-    val subOrganizationIdIndex: Int? = null,
-)
-
-fun getResourceHrnFunc(templateInputs: List<HrnTemplateInput>): (ApplicationRequest) -> ResourceHrn {
+fun getResourceHrnFunc(templateInputs: List<RouteOption>): (ApplicationRequest) -> ResourceHrn {
```

Comment:
In the new model, are we passing `List<RouteOption>` anywhere? If not, we can make this function accept just one element.